File: src/main/java/com/example/dpop/orchestrator/account/AccountBindingKeyMappingRepository.java
package com.example.dpop.orchestrator.account;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface AccountBindingKeyMappingRepository extends JpaRepository<AccountBindingKeyMapping, String> {

    Optional<AccountBindingKeyMapping> findByBindingKeyRef(String bindingKeyRef);
}

File: src/main/kotlin/com/example/dpop/orchestrator/account/AccountBindingKeyMappingRepository.kt
package com.example.dpop.orchestrator.account

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface AccountBindingKeyMappingRepository : JpaRepository<AccountBindingKeyMapping, String> {

    fun findByBindingKeyRef(bindingKeyRef: String): AccountBindingKeyMapping?
}
